code/modules/mob/living/silicon/robot/robot_items.dm

```diff
@@ -294,26 +294,36 @@
 		pick_up(A, user)
 
 /obj/item/weapon/inflatable_dispenser/proc/try_deploy_inflatable(var/turf/T, var/mob/living/user)
+	if (deploying)
+		return
+	deploying = 1
+	var/newtype
 	if(mode) // Door deployment
 		if(!stored_doors)
 			user << "\The [src] is out of doors!"
 			return
 
 		if(T && istype(T))
-			new /obj/structure/inflatable/door(T)
-			stored_doors--
+			newtype = /obj/structure/inflatable/door
 
 	else // Wall deployment
 		if(!stored_walls)
 			user << "\The [src] is out of walls!"
 			return
 
 		if(T && istype(T))
-			new /obj/structure/inflatable/wall(T)
-			stored_walls--
+			newtype = /obj/structure/inflatable/wall
 
+	user.visible_message(span("notice", "[user] starts deploying an inflatable"), span("notice", "You start deploying an inflatable [mode ? "door" : "wall"]!"))
 	playsound(T, 'sound/items/zip.ogg', 75, 1)
-	user << "You deploy the inflatable [mode ? "door" : "wall"]!"
+	if (do_after(user, 20, needhand = 0))
```

Comment:
We had this discussion already, skull said to do it.

Its only two seconds, it improves the feel of things honestly, try it.